src/Adapter/Presenter/Product/ProductLazyArray.php

```diff
@@ -541,49 +541,57 @@ public function getMainVariants()
     }
 
     /**
+     * Returns combination specific data, if assigned. This function should be rewritten because it
+     * loads the data from the first attribute found. See ProductController for more info.
+     *
+     * Also, on product page, $this->product['attributes'] contains a list of combinations, while in cart
+     * it contains only attribute pairs like Color-Black etc.
+     *
      * @arrayAccess
      *
      * @return array|null
      */
-    public function getSpecificReferences()
+    public function getCombinationSpecificData()
     {
-        if (isset($this->product['cart_quantity'])) {
+        if (!isset($this->product['attributes']) || empty($this->product['attributes'])) {
             return null;
         }
 
-        // If the product has no combinations then the `specific_references` must be filled in
-        if (isset($this->product['attributes'])) {
-            $specificReferences = array_slice($this->product['attributes'], 0)[0];
-        } else {
-            $specificReferences = [
-                'isbn' => $this->product['isbn'] ?? false,
-                'upc' => $this->product['upc'] ?? false,
-                'ean13' => $this->product['ean13'] ?? false,
-                'mpn' => $this->product['mpn'] ?? false,
-            ];
+        $data = array_slice($this->product['attributes'], 0);
+
+        return isset($data[0]) ? $data[0] : null;
```

Comment:
As `$this->product['attributes']` cannot be empty because you checked it before, you can simply do:
```suggestion

        return reset($this->product['attributes'])
```

Reply:
done